diablo/api/blackout_controller.py

```diff
@@ -0,0 +1,101 @@
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE. THE
+SOFTWARE AND ACCOMPANYING DOCUMENTATION, IF ANY, PROVIDED HEREUNDER IS PROVIDED
+"AS IS". REGENTS HAS NO OBLIGATION TO PROVIDE MAINTENANCE, SUPPORT, UPDATES,
+ENHANCEMENTS, OR MODIFICATIONS.
+"""
+from diablo.api.errors import BadRequestError, ResourceNotFoundError
+from diablo.api.util import admin_required
+from diablo.lib.http import tolerant_jsonify
+from diablo.lib.util import localized_timestamp_to_utc
+from diablo.models.blackout import Blackout
+from flask import current_app as app, request
+
+
+@app.route('/api/blackouts/all')
+@admin_required
+def get_all_blackouts():
+    return tolerant_jsonify([blackout.to_api_json() for blackout in Blackout.all_blackouts()])
+
+
+@app.route('/api/blackout/<blackout_id>')
+@admin_required
+def get_blackout(blackout_id):
+    blackout = Blackout.get_blackout(blackout_id)
+    if blackout:
+        return tolerant_jsonify(blackout.to_api_json())
+    else:
+        raise ResourceNotFoundError('No such blackout')
+
+
+@app.route('/api/blackout/<blackout_id>', methods=['DELETE'])
+@admin_required
+def delete_blackout(blackout_id):
+    Blackout.delete_blackout(blackout_id)
+    return tolerant_jsonify({'message': f'Email template {blackout_id} has been deleted'}), 200
+
+
+@app.route('/api/blackout/create', methods=['POST'])
+@admin_required
+def create_blackout():
+    params = request.get_json()
+    name = params.get('name')
+    start_date = params.get('startDate')
+    end_date = params.get('endDate')
+
+    if None in [name, start_date, end_date]:
+        raise BadRequestError('Required parameters are missing.')
+
+    blackout = Blackout.create(
+        name=name,
+        start_date=_local_blackout_date_to_utc(start_date),
+        end_date=_local_blackout_date_to_utc(end_date),
+    )
+    return tolerant_jsonify(blackout.to_api_json())
+
+
+@app.route('/api/blackout/update', methods=['POST'])
+@admin_required
+def update_blackout():
+    params = request.get_json()
+    blackout_id = params.get('blackoutId')
+    blackout = Blackout.get_blackout(blackout_id) if blackout_id else None
+    if blackout:
+        name = params.get('name')
+        start_date = params.get('startDate')
+        end_date = params.get('endDate')
+
+        if None in [name, start_date, end_date]:
+            raise BadRequestError('Required parameters are missing.')
+
+        blackout = Blackout.update(
+            blackout_id=blackout_id,
+            name=name,
+            start_date=_local_blackout_date_to_utc(start_date),
+            end_date=_local_blackout_date_to_utc(end_date),
+        )
+        return tolerant_jsonify(blackout.to_api_json())
+    else:
+        raise ResourceNotFoundError('No such email template')
+
+
+def _local_blackout_date_to_utc(blackout_date):
+    return localized_timestamp_to_utc(blackout_date, date_format='%Y-%m-%d')
```

Comment:
@johncrossman Should we raise a BadRequestError here if the user-supplied date param doesn't parse correctly?

Reply:
@pauline2k - Yes, I agree.